src/Datadog.Trace/TraceContext.cs

```diff
@@ -46,6 +46,7 @@ public void CloseSpan(Span span)
                 if (_openSpans == 0)
                 {
                     _tracer.Write(_spans);
+                    _spans = new List<Span>();
```

Comment:
Does that make sense? It would mean creating a child for an already closed span right?
(If yes, I think it would be useful to add a simple test exercising this code path)